GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+
+namespace GenericsHomework
+{
+    public class Node<T>
+    {
+        public Node(T init) 
+        {
+            this.data = init;
+            this.Next = this;
+        }
+        
+        public Node()
+        {
+            this.Next = this;
+        }
+
+        public T? data;
+        public Node<T> Next { get; private set; }

```

Comment:
```suggestion
        public Node<T> Next { get; private set; }
```
This shouldn't be an auto property, this should be a property with its `get` and `set` implemented.